tools/boot/install-boot.sh

```diff
@@ -95,6 +100,27 @@ make_esp_device() {
 
     efibootname=$(get_uefi_bootname)
     kbfree=$(df -k "${mntpt}" | tail -1 | cut -w -f 4)
+
+    # For amd64 there are two efi loaders, one for 64-bit uefi and the other for 32-bit.
+    # If we can figure out which one we need, we copy it to /EFI/freebsd/loader.efi
+    # If we can't, then install the 64-bit one as loader.efi and the 32-bit one as loader_ia32.efi
+    if ([ -n "${updatesystem}" ] && [ `sysctl -n machdep.efi_arch` == i386 ]) || \
+       ([ -f "${mntpt}/EFI/freebsd/loader.efi" ] && (file "${mntpt}/EFI/freebsd/loader.efi" | grep -q '80386')) ; then
+        file="${file%.efi}_ia32.efi"
```

Comment:
As above

Reply:
Maybe we should make it so the 64-bit one is always loader.efi and the 32-bit one is always loader_ia32.efi

Looking at this now, I think occasionally copying the 32-bit one to loader.efi would just cause confusion.